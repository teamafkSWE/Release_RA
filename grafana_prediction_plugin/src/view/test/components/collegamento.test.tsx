import React from 'react'
// add custom jest matchers from jest-dom
import '@testing-library/jest-dom/extend-expect'
// the component to test
import Collegamento from "../../components/collegamento";
import {cleanup, render, fireEvent, screen} from "@testing-library/react";


describe('Collegamento fc', () => {
    const modify = jest.fn()
    const remove = jest.fn()
    const links = [{predictor: "x", query: "cpu"}, {predictor: 'y', query: "gpu"}]

    beforeEach(() => {
        modify.mockClear()
        remove.mockClear()
    })

    afterEach(() => {
        cleanup()
    })

    test('render', () => {
        const {asFragment} = render(<Collegamento id='1' links={links} nome={'name'} onModify={modify} onRemove={remove}/>)
        expect(asFragment()).toMatchSnapshot()
    })

    test('modify call', () => {
        render(<Collegamento id='1' links={links} nome={'name'} onModify={modify} onRemove={remove}/>)
        fireEvent.click(screen.getByText('Edit'))
        expect(modify).toBeCalledTimes(1)
        expect(modify).toBeCalledWith('1')
    })

    test('remove call', () => {
        render(<Collegamento id='1' links={links} nome={'name'} onModify={modify} onRemove={remove}/>)
        fireEvent.click(screen.getByText('Disconnect'))
        expect(remove).toBeCalledTimes(1)
        expect(remove).toBeCalledWith('1')
    })
})