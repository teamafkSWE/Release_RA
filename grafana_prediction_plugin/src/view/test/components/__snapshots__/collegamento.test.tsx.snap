// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collegamento fc render 1`] = `
<DocumentFragment>
  <div
    style="width: 100%; border: 1pt solid white; border-radius: 3pt; min-width: 12rem; margin-top: .8rem;"
  >
    <p
      style="text-align: center; border-bottom: 1pt solid white; margin: 0px; padding: 0.4rem 0px 0.4rem; font-size: 1.2rem;"
    >
      name
    </p>
    <div
      style="display: flex; justify-content: space-evenly;"
    >
      <div
        style="width: 100%;"
      >
        <p
          style="text-align: center; width: 100%; margin: 0px; padding: 0.4rem 0.6rem 0.4rem; border-right: 1pt solid white; border-bottom: 1pt solid white;"
        >
          Predictor
        </p>
        <p
          style="text-align: center; width: 100%; margin: 0px; padding: 0.2rem 0.6rem 0.2rem;"
        >
          x
        </p>
        <p
          style="text-align: center; width: 100%; margin: 0px; padding: 0.2rem 0.6rem 0.2rem;"
        >
          y
        </p>
      </div>
      <div
        style="width: 100%;"
      >
        <p
          style="text-align: center; width: 100%; margin: 0px; padding: 0.4rem 0.6rem 0.4rem; border-bottom: 1pt solid white;"
        >
          Node
        </p>
        <p
          style="text-align: center; width: 100%; margin: 0px; padding: 0.2rem 0.6rem 0.2rem;"
        >
          cpu
        </p>
        <p
          style="text-align: center; width: 100%; margin: 0px; padding: 0.2rem 0.6rem 0.2rem;"
        >
          gpu
        </p>
      </div>
    </div>
  </div>
  <div
    style="display: flex; justify-content: space-evenly; width: 100%; margin: .75rem 0px .5rem;"
  >
    <button
      class="css-1ihbihm-button"
      style="margin: 0px .5rem 0px;"
    >
      <span
        class="css-1beih13"
      >
        <span>
          Edit
        </span>
      </span>
    </button>
    <button
      class="css-1ihbihm-button"
      style="margin: 0px .5rem 0px;"
    >
      <span
        class="css-1beih13"
      >
        <span>
          Disconnect
        </span>
      </span>
    </button>
  </div>
</DocumentFragment>
`;
